app/main/views/communications.py

```diff
@@ -1,37 +1,69 @@
+from pathlib import PurePath
+
 from dmutils import s3  # this style of import so we only have to mock once
-from dmutils.documents import file_is_pdf, file_is_csv, file_is_open_document_format
+from dmutils.documents import file_is_pdf, file_is_csv, file_is_open_document_format, get_signed_url
 from dmutils.flask import timed_render_template as render_template
-from flask import redirect, url_for, current_app, request, flash
+from flask import redirect, url_for, current_app, request, flash, abort
 
 from .. import main
 from ..auth import role_required
 from ... import data_api_client
 
 
+def _get_comm_type_root(framework_slug, comm_type):
+    # PurePath has the advantage that it doesn't handle ".." elements specially, and so we're safe from a whole
+    # category of tricks
+    return PurePath(framework_slug) / "communications" / "updates" / f"{comm_type}s"
+
+
+_comm_types = ("communication", "clarification",)
+
+
 @main.route('/communications/<framework_slug>', methods=['GET'])
 @role_required('admin-framework-manager')
 def manage_communications(framework_slug):
     communications_bucket = s3.S3(current_app.config['DM_COMMUNICATIONS_BUCKET'])
     framework = data_api_client.get_framework(framework_slug)['frameworks']
 
-    # Get the last items from the timestamp-ordered buckets, in order to show the most recently updated timestamps
-    # The "or [None]" ensures we have a list with at least one item (None) when no files exist in the bucket
-    communications = communications_bucket.list(
-        '{}/communications/updates/communications'.format(framework_slug), load_timestamps=True
-    ) or [None]
-    clarifications = communications_bucket.list(
-        '{}/communications/updates/clarifications'.format(framework_slug), load_timestamps=True
-    ) or [None]
+    # generate a dict of comm_type: seq of s3 object dicts
+    comm_type_objs = {
+        comm_type: tuple(
+            {
+                **obj_dict,
+                # annotate on to object dicts their paths relative to comm_type_root
+                "rel_path": PurePath(obj_dict["path"]).relative_to(_get_comm_type_root(framework_slug, comm_type)),
+            } for obj_dict in communications_bucket.list(
```

Comment:
I feel like renaming `obj_dict` to `bucket_object` or `bucket_item` would make it a bit easier to follow what's going on here.

```suggestion
            } for bucket_listing in communications_bucket.list(
```

Reply:
Guess so